# Exercício 02 — Consumo de combustível em rota

**Nome:** Wagner Fadiga

## Descrição

Neste exercício será analisado o consumo de combustível durante uma rota.
O programa irá percorrer as etapas da viagem e calcular o consumo total de combustível.
Também será apresentado o consumo médio da rota.

In [3]:
# Dados da rota
distancias_km = [120, 85, 150, 95]
consumo_km_l = 12

In [4]:
# Cálculo do consumo em cada etapa
consumos = []

for distancia in distancias_km:
    consumo = distancia / consumo_km_l
    consumos.append(consumo)

print("Consumo em cada etapa:")
for i, consumo in enumerate(consumos, start=1):
    print(f"Etapa {i}: {consumo:.2f} litros")

Consumo em cada etapa:
Etapa 1: 10.00 litros
Etapa 2: 7.08 litros
Etapa 3: 12.50 litros
Etapa 4: 7.92 litros


In [5]:
# Cálculo do consumo total e da distância total
distancia_total = sum(distancias_km)
consumo_total = sum(consumos)

print(f"Distância total: {distancia_total:.2f} km")
print(f"Consumo total: {consumo_total:.2f} litros")

Distância total: 450.00 km
Consumo total: 37.50 litros


In [6]:
# Cálculo do consumo médio por etapa
consumo_medio = consumo_total / len(distancias_km)

print(f"Consumo médio por etapa: {consumo_medio:.2f} litros")

Consumo médio por etapa: 9.38 litros


In [7]:
# Relatório final da rota
print("=== RELATÓRIO DE CONSUMO ===")
print(f"Distância total: {distancia_total:.2f} km")
print(f"Consumo total: {consumo_total:.2f} litros")
print(f"Consumo médio por etapa: {consumo_medio:.2f} litros")
print("============================")

=== RELATÓRIO DE CONSUMO ===
Distância total: 450.00 km
Consumo total: 37.50 litros
Consumo médio por etapa: 9.38 litros


## Interpretação do resultado

A rota possui uma distância total de 450 km e um consumo total de 37,50 litros.
O consumo médio por etapa permite acompanhar quanto combustível foi utilizado em cada trecho da viagem.
Os resultados ajudam a analisar o consumo de combustível durante o percurso.